### PR TITLE
tapdb: fix multiple issues around migration file handling

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightninglabs/taproot-assets/issues/1071.

This PR is the result after almost two days of debugging #1071.
The TL;DR is: due to a git branch cherry-pick mishap, the migration number 14 wasn't included in the `v0.3.3` release version. Some users then applied the migration directly as it was in the source directory, which lead to them not having properly working auto incrementing primary keys.

The first commit in this PR adds a CI check that verifies migration numbers in the file names are continuous.

The second commit re-applies the migration 14 (somewhat modified) as migration 23.

The last commit then fixes the workaround with the primary key type replacements we used. See comment in `scripts/gen_sqlc_docker.sh` if you want to understand the full problem and solution.

### CODE REVIEW

**Review by ffranr — APPROVED** [review]
Looks good :+1: I've tested the script locally on Linux.

**Comment by ffranr** [comment]
Another fee estimation itest here:
```
--- FAIL: TestTaprootAssetsDaemon/fee_estimation (16.48s)
```

File: scripts/check-migration-numbering.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+migrations_path="tapdb/sqlc/migrations"
+
+# Check if the directory exists
+if [ ! -d "$migrations_path" ]; then
+  echo "Directory $migrations_path does not exist."
+  exit 1
+fi
+
+# Get all unique prefixes (e.g., 000001, always 6 digits) from .up.sql files.
+prefixes=($(ls "$migrations_path"/*.up.sql 2>/dev/null | \
```
Comment (jharveyb) [inline scripts/check-migration-numbering.sh:13]: Super-nit; a simpler command:

`ls -1 tapdb/sqlc/migrations/*.up.sql | cut -d'_' -f1 | cut -d'/' -f4 | sort -n`

**Review by jharveyb — APPROVED** [review]
LGTM!

Tested the script with a fake migration of prefix `000025`, both only with `up` and `up`/`down`.

Also checked on the `BIGINT PRIMARY KEY` -> `INTEGER PRIMARY KEY` replacement, and that `BIGSERIAL` is the right Postgres type to use here.

File: scripts/check-migration-numbering.sh (hunk 1 above)
Comment (guggero) [reply scripts/check-migration-numbering.sh:13]: That seems a bit brittle if someone uses unexpected characters in the file name...

**Comment by guggero** [comment]
Going to verify the postgres itest suceeds locally before merging. I'm also going to build a `litd` image with this and test it against my own node.

**Comment by guggero** [comment]
> Going to verify the postgres itest suceeds locally before merging. I'm also going to build a `litd` image with this and test it against my own node.

Applied this to my node without problems and also ran the full postgres itest locally.
